lib/src/analysis/lte_downgrade.rs

```diff
@@ -2,80 +2,41 @@ use std::borrow::Cow;
+        Cow::from("Tests if a cell releases our connection and redirects us to a 2G cell.")
     }
 
-    fn analyze_information_element(&mut self, ie: &InformationElement) -> Option<super::analyzer::Event> {
-        let sibs = &self.unpack_system_information(ie)?.0;
-        for sib in sibs {
-            match sib {
-                SystemInformation_r8_IEsSib_TypeAndInfo_Entry::Sib6(sib6) => {
-                    if let Some(carrier_info_list) = sib6.carrier_freq_list_utra_fdd.as_ref() {
-                        for carrier_info in &carrier_info_list.0 {
-                            if let Some(CellReselectionPriority(p)) = carrier_info.cell_reselection_priority {
-                                if p == 0 {
-                                    return Some(Event {
-                                        event_type: EventType::QualitativeWarning { severity: Severity::High },
-                                        message: "LTE cell advertised a 3G cell for priority 0 reselection".to_string(),
-                                    });
-                                }
-                            }
-                        }
-                    }
-                    if let Some(carrier_info_list) = sib6.carrier_freq_list_utra_tdd.as_ref() {
-                        for carrier_info in &carrier_info_list.0 {
-                            if let Some(CellReselectionPriority(p)) = carrier_info.cell_reselection_priority {
-                                if p == 0 {
-                                    return Some(Event {
-                                        event_type: EventType::QualitativeWarning { severity: Severity::High },
-                                        message: "LTE cell advertised a 3G cell for priority 0 reselection".to_string(),
-                                    });
-                                }
-                            }
-                        }
-                    }
-                },
-                SystemInformation_r8_IEsSib_TypeAndInfo_Entry::Sib7(SystemInformationBlockType7 { carrier_freqs_info_list: Some(carrier_info_list), .. }) => {
-                    for carrier_info in &carrier_info_list.0 {
-                        if let Some(CellReselectionPriority(p)) = carrier_info.common_info.cell_reselection_priority {
-                            if p == 0 {
-                                return Some(Event {
-                                    event_type: EventType::QualitativeWarning { severity: Severity::High },
-                                    message: "LTE cell advertised a 2G cell for priority 0 reselection".to_string(),
-                                });
-                            }
-                        }
-                    }
-                },
-                _ => {},
-            }
+    fn analyze_information_element(&mut self, ie: &InformationElement) -> Option<Event> {
```

Comment:
I don't think we should get rid of our old heuristic, I think it is still useful to have! We should just add this as an additional heuristic. Lets leave the old one in!

Reply:
good point!